code/modules/mining/equipment/kheiral_cuffs.dm

```diff
@@ -0,0 +1,129 @@
+/// Acts as a GPS beacon & distress signaller when clamped to one's wrist off-station.
+/obj/item/kheiral_cuffs
+	name = "\improper Kheiral cuffs"
+	desc = "A prototype wrist communicator powered by Kheiral Matter. When both ends are clamped to one wrist, can send a distress signal upon your death off-station.\nA small engraving on the inside reads, \"NOT HANDCUFFS\""
+	icon = 'icons/obj/mining.dmi'
+	icon_state = "strand"
+	worn_icon_state = "strandcuff"
+	slot_flags = ITEM_SLOT_GLOVES
+	throwforce = 0
+	w_class = WEIGHT_CLASS_SMALL
+	gender = PLURAL
+	throw_speed = 3
+	throw_range = 5
+	/// If we're in the glove slot
+	var/on_wrist = FALSE
+	/// If the GPS is already on
+	var/gps_enabled = FALSE
+	/// If we're off the station's Z-level
+	var/far_from_home = FALSE
+	/// Used to broadcast user's death through the radio
+	var/obj/item/radio/internal_radio
+	/// The GPS component used by the cuffs. Extremely unoptimal.
+	var/datum/component/gps/gps
+
+/obj/item/kheiral_cuffs/Initialize()
+	. = ..()
+	update_icon(UPDATE_OVERLAYS)
+	RegisterSignal(src, COMSIG_MOVABLE_Z_CHANGED, .proc/check_z)
+
+	internal_radio = new/obj/item/radio(src)
+	internal_radio.subspace_transmission = TRUE
+	internal_radio.canhear_range = 0
+	internal_radio.should_be_listening = FALSE
+	internal_radio.keyslot = new /obj/item/encryptionkey/strandcuffs
+	internal_radio.set_listening(FALSE)
+	internal_radio.recalculateChannels()
+
+	AddComponent(/datum/component/gps, "Kheiral Link", FALSE)
+	gps = GetComponent(/datum/component/gps)
+
+/obj/item/kheiral_cuffs/examine(mob/user)
+	. = ..()
+	if(gps_enabled)
+		. += span_notice("The cuff's GPS signal is on.")
+
+/obj/item/kheiral_cuffs/item_action_slot_check(slot)
+	return slot == ITEM_SLOT_GLOVES
+
+/obj/item/kheiral_cuffs/equipped(mob/user, slot, initial)
+	. = ..()
+	if(slot != ITEM_SLOT_GLOVES)
+		return
+	on_wrist = TRUE
+	playsound(loc, 'sound/weapons/handcuffs.ogg', 30, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	connect_kheiral_network(user)
+
+/obj/item/kheiral_cuffs/dropped(mob/user, silent)
+	. = ..()
+	if(on_wrist)
+		playsound(loc, 'sound/weapons/handcuffs.ogg', 30, TRUE, SHORT_RANGE_SOUND_EXTRARANGE)
+	on_wrist = FALSE
+	remove_kheiral_network(user)
+
+/// Enables the GPS & registers the statchange signal
+/obj/item/kheiral_cuffs/proc/connect_kheiral_network(mob/user)
+	if(gps_enabled)
+		return
+	if(on_wrist && far_from_home)
+		gps.gpstag = "*[user.name]'s Kheiral Link"
+		gps.tracking = TRUE
+		balloon_alert(user, "GPS activated")
+		RegisterSignal(user, COMSIG_MOB_STATCHANGE, .proc/on_user_statchange)
+		gps_enabled = TRUE
+
+/// Disables the GPS and unregisters the statchange signal
+/obj/item/kheiral_cuffs/proc/remove_kheiral_network(mob/user)
+	if(!gps_enabled)
+		return
+	if(!on_wrist || !far_from_home)
+		gps.tracking = FALSE
+		balloon_alert(user, "GPS de-activated")
+		UnregisterSignal(user, COMSIG_MOB_STATCHANGE)
+		gps_enabled = FALSE
+
+/obj/item/kheiral_cuffs/Destroy(force)
+	. = ..()
```

Comment:
It's typically best to do destroy returns at the end, rather then at the start.
It's often not a problem, but things can get really headachey if someone improperly gives some action you do side effects, and accidentially say, registers a signal onto us **after** we've cleared all our signals